File: ios/Packages/SolennixCore/Sources/SolennixCore/Models/EventProduct.swift
import Foundation

public struct EventProduct: Codable, Identifiable, Sendable, Hashable {
    public var id: String
    public var eventId: String
    public var productId: String
    public var quantity: Int
    public var unitPrice: Double
    public var discount: Double
    public var totalPrice: Double?
    public var createdAt: String

    public init(
        id: String, eventId: String, productId: String, quantity: Int,
        unitPrice: Double, discount: Double, totalPrice: Double? = nil, createdAt: String
    ) {
        self.id = id; self.eventId = eventId; self.productId = productId
        self.quantity = quantity; self.unitPrice = unitPrice; self.discount = discount
        self.totalPrice = totalPrice; self.createdAt = createdAt
    }

    enum CodingKeys: String, CodingKey {
        case id
        case eventId = "event_id"
        case productId = "product_id"
        case quantity
        case unitPrice = "unit_price"
        case discount
        case totalPrice = "total_price"
        case createdAt = "created_at"
    }
}

File: android/core/model/src/main/java/com/creapolis/solennix/core/model/EventProduct.kt
package com.creapolis.solennix.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class EventProduct(
    val id: String,
    @SerialName("event_id") val eventId: String,
    @SerialName("product_id") val productId: String,
    val quantity: Double,
    @SerialName("unit_price") val unitPrice: Double,
    val discount: Double = 0.0,
    @SerialName("total_price") val totalPrice: Double? = null,
    @SerialName("created_at") val createdAt: String = ""
)
